internal/expr/prepare.go

```diff
@@ -56,28 +56,139 @@ func starCount(fns []fullName) int {
 }
 
 // prepareInput checks that the input expression corresponds to a known type.
-func prepareInput(ti typeNameToInfo, p *inputPart) (typeMember, error) {
-	info, ok := ti[p.sourceType.prefix]
-	if !ok {
-		ts := getKeys(ti)
-		if len(ts) == 0 {
-			return nil, fmt.Errorf(`type %q not passed as a parameter`, p.sourceType.prefix)
-		} else {
-			return nil, fmt.Errorf(`type %q not passed as a parameter, have: %s`, p.sourceType.prefix, strings.Join(ts, ", "))
+func prepareInput(ti typeNameToInfo, p *inputPart) ([]fullName, []typeMember, error) {
+	var inCols = make([]fullName, 0)
+	var typeMembers = make([]typeMember, 0)
```

Comment:
Just for consistency, could these use `:=` like the block below? Also, with these values declared before the helper closures, it looks like they should be used in them, but this doesn't appear to be the case. Could their declaration be moved later?

Reply:
I never know when to use `:=` vs `var`, it really seems to just be a matter of personal preference.

I've moved the declarations down to just before where they are used. The reason those are at the top is because they are the return values, so I'll just move them to be named return values.